lib/src/util/component_usage_migrator.dart

```diff
@@ -0,0 +1,791 @@
+        end = child.end;
+      }
+    }
+    yieldPatch('', start, end);
+  }
+
+  /// Yields a patch that updates a prop, updating either its name ([newName]),
+  /// its right-hand side ([newRhs]), or both, and optionally adding an
+  /// [additionalCascadeSection].
+  ///
+  /// This method makes it convenient to update props without dealing with offsets.
+  void yieldPropPatch(
+    PropAssignment prop, {
+    String? newName,
+    String? newRhs,
+    String? additionalCascadeSection,
+  }) {
+    if (newName == null && newRhs == null) {
+      throw ArgumentError.notNull('either newName or newRhs');
+    }
+
+    if (newName != null) {
+      yieldPatchOverNode(newName, prop.name);
+    }
+
+    if (newRhs != null) {
+      yieldPatchOverNode(newRhs, prop.rightHandSide);
+    }
+
+    if (additionalCascadeSection != null) {
+      // Add spaces so that dartfmt has a better time in case the cascade section has leading line comments
+      yieldInsertionPatch(
+          '\n  $additionalCascadeSection', prop.rightHandSide.end);
+    }
+  }
+
+  /// The prefix for FIX-ME comments added in the various yield* methods in this class.
+  ///
+  /// E.g., `'FIXME'`, `'FIXME(something_more_specific)'`.
+  String get fixmePrefix;
+
+  /// Yields a patch with a fix-me comment before a given component [usage]
+  /// with a custom [message].
+  void yieldUsageFixmePatch(FluentComponentUsage usage, String message) {
+    yieldInsertionPatch(
+        lineComment('$fixmePrefix $message'), usage.node.offset);
+  }
+
+  void yieldPropManualMigratePatch(PropAssignment prop) {
+    yieldPropFixmePatch(prop, 'manually migrate');
+  }
+
+  /// Yields a patch with a fix-me comment before a given [prop]
+  /// with a custom [message].
+  ///
+  /// Inserts extra newlines as necessary to avoid the comment getting "stuck"
+  /// to the previous line, or adding blank lines.
+  ///
+  /// For example:
+  /// ```dart
+  /// // Flagging "id" for this usage...
+  /// (Dom.div()..id = '')();
+  ///
+  /// // ...results in this...
+  /// (Dom.div()
+  ///   // FIXME ...
+  ///   ..id = ''
+  /// )();
+  ///
+  /// // ...and not this...
+  /// (Dom.div() // FIXME ...
+  ///   ..id = ''
+  /// )();
+  /// ```
+  void yieldPropFixmePatch(PropAssignment prop, String message) {
+    yieldBuilderMemberFixmePatch(prop, '${prop.name.name} prop - $message');
+  }
+
+  /// Yields a patch with a fix-me comment before a given [access]
+  /// with a custom [message].
+  ///
+  /// Inserts extra newlines as necessary to avoid the comment getting "stuck"
+  /// to the previous line, or adding blank lines.
+  ///
+  /// For example:
+  /// ```dart
+  /// // Flagging "id" for this usage...
+  /// (Dom.div()..id = '')();
+  ///
+  /// // ...results in this...
+  /// (Dom.div()
+  ///   // FIXME ...
+  ///   ..id = ''
+  /// )();
+  ///
+  /// // ...and not this...
+  /// (Dom.div() // FIXME ...
+  ///   ..id = ''
+  /// )();
+  void yieldBuilderMemberFixmePatch(
```

Comment:
The example code block isn't terminated
```suggestion
  /// ```dart
  /// // Flagging "id" for this usage...
  /// (Dom.div()..id = '')();
  ///
  /// // ...results in this...
  /// (Dom.div()
  ///   // FIXME ...
  ///   ..id = ''
  /// )();
  ///
  /// // ...and not this...
  /// (Dom.div() // FIXME ...
  ///   ..id = ''
  /// )();
  /// ```
  void yieldBuilderMemberFixmePatch(
```